korio/src/jvmMain/kotlin/com/soywiz/korio/file/std/LocalVfsJvm.kt

```diff
@@ -402,22 +402,16 @@ private class LocalVfsJvm : LocalVfsV2() {
 		val fs = FileSystems.getDefault()
 		val watcher = fs.newWatchService()
 
-		fs.getPath(path).register(
+        val registeredKey = fs.getPath(path).register(
 			watcher,
 			StandardWatchEventKinds.ENTRY_CREATE,
 			StandardWatchEventKinds.ENTRY_DELETE,
 			StandardWatchEventKinds.ENTRY_MODIFY
 		)
 
-		launchImmediately(coroutineContext) {
+        GlobalScope.launch(Dispatchers.IO) {
 			while (running) {
-				val key = executeIo {
-					var r: WatchKey?
-					do {
-						r = watcher.poll(100L, TimeUnit.MILLISECONDS)
-					} while (r == null && running)
-					r
-				} ?: continue
+                val key = watcher.take()
 
 				for (e in key.pollEvents()) {
 					val kind = e.kind()
```

Comment:
Could you wrap `when (kind) { ... }` with `withContext(coroutineContext) { when (kind) { ... } }` that way handler will be executed in the right thread where the function was called, in korio and korge this is important since it is mostly single threaded and mostly not thread-safe